lib/charms/mongodb/v1/mongodb_provider.py

```diff
@@ -76,8 +76,8 @@ def __init__(self, charm: CharmBase, substrate="k8s", relation_name: str = REL_N
             self.database_provides.on.database_requested, self._on_relation_event
         )
 
-    def pass_hook_checks(self, event: EventBase) -> bool:
-        """Runs the pre-hooks checks for MongoDBProvider, returns True if all pass."""
+    def sanity_hook_cheks(self) -> bool:
```

Comment:
1. Could you rename this method with the `is_` prefix? it is unclear from the name what this check does. 
2. Why don't we extend the existing `pass_hook_checks` and call it everywhere instead of a second method?

Reply:
The `pass_hook_checks` has two parts:
 * The first one which I extracted to the `sanity_hook_checks` which is event independent
 * The second part is event dependent and can defer the event.
 
 In this case, we don't want to defer the `update_status` event, we just want to protect the call.
 That's why I decided to split the hook checks between the sanity ones which always have to be run, and the ones that can defer the event.
The existing `pass_hook_checks` hasn't changed it what it does, I just went for reusability :)

Reply:
> 1. Could you rename this method with the `is_` prefix? it is unclear from the name what this check does.

I think I'll go to `pass_sanity_hook_checks` which is a method name that we already have in other libraries, this way we'll keep some uniformity here.